src/main/java/org/broadinstitute/hellbender/engine/filters/PlatformReadFilter.java

```diff
@@ -5,17 +5,24 @@
 import org.broadinstitute.hellbender.utils.read.GATKRead;
 import org.broadinstitute.hellbender.utils.read.ReadUtils;
 
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.LinkedHashSet;
+import java.util.List;
 import java.util.Set;
 
 /**
  * Keep only reads that match th PL attribute.
  * Matching is done by case-insensitive substring matching
  * (checking if the read's platform tag contains the given string).
  */
-public final class PlatformReadFilter implements ReadFilter {
+public final class PlatformReadFilter implements ReadFilter, CommandLineFilter, Serializable {
     private static final long serialVersionUID = 1L;
-    @Argument(fullName = "PLFilterName", shortName = "PLFilterName", doc="Keep reads with RG:PL attribute containing this string", optional=true)
-    public Set<String> PLFilterNames;
+
+    private final static String platformArgName = "PLFilterName";
+
+    @Argument(fullName = platformArgName, shortName = "PLFilterName", doc="Keep reads with RG:PL attribute containing this string", optional=true)
```

Comment:
@lbergelson If I make this argument optional=false, then its ALWAYS required, whether you specify the corresponding filter or not (I haven't been able to reproduce the bug I mentioned the other day). Instead, its optional=true, but contained inside of an argument collection that is included with a dependsOnArgument="RF" and dependsOnValue=PLPlatformFilter", which gives the right result (it complains only when you set the argument value but don't specify the filter, or when you specify the filter but don't set the filter argument).